VirtoCommerce.MarketingModule.Data/Observers/CouponUsageRecordObserver.cs

```diff
@@ -27,28 +27,28 @@ public CouponUsageRecordObserver(IPromotionUsageService usageService)
 
         public void OnNext(OrderChangedEvent changedEvent)
         {
-            if (changedEvent.ChangeState == EntryState.Added)
-            {
-                var oldUsages = new List<PromotionUsage>();
-                var newUsages = GetCouponUsages(changedEvent.ModifiedOrder.Id, changedEvent.ModifiedOrder);               
-                RecordUsages(changedEvent.ModifiedOrder.Id, oldUsages, newUsages);
-            }
+            //if (changedEvent.ChangeState == EntryState.Added)
```

Comment:
Could you convert this Observer to domain event handler?

Reply:
done